Sources/SyntaxAnalysis/DeclarationSyntaxVisitor.swift

```diff
@@ -517,3 +517,64 @@ public final class DeclarationSyntaxVisitor: PeripherySyntaxVisitor {
             }
     }
 }
+
+// MARK: - Inline Comment Commands
+
+private extension SyntaxProtocol {
+    /// All trivia where inline comment commands can be found.
+    ///
+    /// Matches uses like these:
+    ///
+    /// ```
+    /// // periphery:ignore
+    /// Foo {
+    /// }
+    /// ```
+    ///
+    /// ```
+    /// Foo { // periphery:ignore
+    /// }
+    /// ```
+    ///
+    /// ```
+    /// Foo {} // periphery:ignore
+    /// ```
+    var commentCommandTrivia: Trivia {
+        var commandTrivia = leadingTrivia
+
+        if let hasMemberBlock = self as? HasMemberBlock {
+            commandTrivia = commandTrivia
+                .merging(hasMemberBlock.memberBlock.leftBrace.trailingTrivia)
+                .merging(hasMemberBlock.memberBlock.rightBrace.trailingTrivia)
+        }
+
+        if let hasCodeBody = self as? HasCodeBody, let body = hasCodeBody.body {
+            commandTrivia = commandTrivia
+                .merging(body.leftBrace.trailingTrivia)
+                .merging(body.rightBrace.trailingTrivia)
+        }
+
+        return commandTrivia
+    }
+}
+
+/// Identifies types with a MemberBlockSyntax child
+private protocol HasMemberBlock {
```

Comment:
There is a version of this upstream in `SwiftSyntaxBuilder`'s [`HasTrailingMemberDeclBlock`](https://github.com/swiftlang/swift-syntax/blob/main/Sources/SwiftSyntaxBuilder/SyntaxNodeWithBody.swift#L155) that would keep the conformance list updated for us, however it would be a new dependency for `SyntaxAnalysis`.

Is that a tradeoff we want to make?

Reply:
Let's keep it as-is, we can add the dependency in the future if we find copying more things over.